hooks/useAllowanceCheck.tsx

```diff
@@ -25,17 +26,20 @@ export default function useAllowanceCheck(
 
   useEffect(() => {
     const erc20AllowanceRes = erc20AllowanceData as CallResult;
+
     if (
       erc20AllowanceError ||
       (erc20AllowanceRes &&
         erc20AllowanceRes["remaining"].low !== UINT_128_MAX &&
         erc20AllowanceRes["remaining"].high !== UINT_128_MAX)
     ) {
       setNeedsAllowance(true);
+      setCurrentAllowance(BigInt(erc20AllowanceRes["remaining"].low));
```

Comment:
Ici de mon côté ça casse la page d'achat de domaine après plusieurs secondes avec l'erreur: `TypeError: Cannot read properties of undefined (reading 'remaining')`. A cause de la condition notamment (`erc20AllowanceError ||`), qui fait exécuter ce code si erc20AllowanceError est vrai (et donc dans ce scénario erc20AllowanceRes est indéfini).

Je pense que le mieux est de mettre
```tsx
if (erc20AllowanceRes) setCurrentAllowance(BigInt(erc20AllowanceRes["remaining"].low));
```
Pour éviter cette erreur